oop_bai_1_edit.php

```diff
@@ -0,0 +1,257 @@
+	array_push($arrayWorkTime, new WorkTime($member["member_code"],$member["start_datetime"],$member["end_datetime"]));
+}
+$employeeFullTime=[];
+foreach ($listMemberFullTime as $employee) {
+	array_push($employeeFullTime, new Employee($employee["code"],$employee["full_name"],$employee["age"],
+		$employee["gender"],$employee["marital_status"],$employee["total_work_time"],$employee["salary"],
+		$employee["start_work_time"],$employee["work_hour"],$employee["has_lunch_break"]));
+}
+$employeePartTime=[];
+foreach ($listMemberPartTime as $employee) {
+	array_push($employeePartTime, new Employee($employee["code"],$employee["full_name"],$employee["age"],
+		$employee["gender"],$employee["marital_status"],$employee["total_work_time"],$employee["salary"],
+		$employee["start_work_time"],$employee["work_hour"],$employee["has_lunch_break"]));
+}
+/*-------------------------------------*/
+class General
+{
+	function dateInMonth($year,$month)
+	{
+		return cal_days_in_month(CAL_GREGORIAN, $month, $year);
+	}
+//========ham tra lai ngay nghi
+	function weekDay($date)
+	{
+		$dayNow = getdate(strtotime($date));
+		if ($dayNow["weekday"] == "Sunday" || $dayNow["weekday"] == "Saturday") {
+			return true;
+		}
+		return false;
+	}
+//==========ham tinh so ngay
+	function dayWorkInMonth($date) 
+    {
+    $dateNow = getdate(strtotime($date));
+    $y = $dateNow["year"];
+    $m = $dateNow["mon"];
+    $holidays = 
+    [
+    	"2019-01-01",
+    	"2019-02-04",
+    	"2019-02-05",
+    	"2019-02-06",
+    	"2019-02-07",
+    	"2019-02-08",
+    	"2019-04-14",
+    	"2019-04-30",
+    	"2019-05-01",
+    	"2019-09-02"
+    ];
+    $holiday = 0;
+    $d = $this->dateInMonth($y, $m);
+    for ($i = 1; $i <= $d; $i++) {
+    	$dayNow= sprintf("%s-%02d-%02d", $y, $m, $i);
+    	if ($this->weekDay($dayNow) || in_array($dayNow, $holidays)) {
+    		$holiday++;
+    	}
+    }
+    return $d - $holiday;
+}
+//===============
+function getTime($data)
+{
+	return date('H:i:s', strtotime($data));
+}
+function hourWorkDay($startTimeDay, $endTimeDay)
+{ 
+	$startTimeDay= $this->getTime($startTimeDay);
+	$endTimeDay = $this->getTime($endTimeDay);
+	return (strtotime($endTimeDay) - strtotime($startTimeDay))/3600;
+}
+function dayMemberFullTime($listWorkTime, $listMemberFullTime, $dataDate)
+{
+	$arDate = getdate(strtotime($dataDate));
+	$monthNeedCal = $arDate["mon"];
+	foreach ($listMemberFullTime as $employee ) {
+		$dayWork = 0;
+		foreach ($listWorkTime as $workTimeDayEmployee) {
+			$monthNow=getdate(strtotime($workTimeDayEmployee->getStartTime()));
+			$monthNow=$monthNow["mon"];
+			$timeWorkDay=0;
+			if($employee->getCode()==$workTimeDayEmployee->getMemberCode()&&$monthNow==$monthNeedCal){                  
+				$startWorkTime = $employee->getStartWorkTime();
+				$inTime = $this->getTime($workTimeDayEmployee->getStartTime()); 
+				$outTime = $this->getTime($workTimeDayEmployee->getEndTime());
+				if (strtotime($startWorkTime) - strtotime($inTime) < 0) {
```

Comment:
if (strtotime($startWorkTime) - strtotime($inTime) < 0) {
					$timeWorkDay = $this->hourWorkDay($inTime, $outTime) - 1.5;
                    } else { // di som
                    	$timeWorkDay = $this->hourWorkDay($startWorkTime, $outTime) - 1.5;
                    }

Đoạn này có mục đích gì?

Reply:
Đoạn này là phần em tính công theo số giờ làm, 
trường hợp 1, nhân viên đi sớm hơn giờ đăng ký làm thì sẽ tính giờ làm từ (giờ đăng ky- lúc về)
trường hợp 2, nhân viên đi sau giờ làm giờ làm sẽ tính từ (giờ đến- giờ về)